File: 001-math/002-Divisor game/Java/Solution.java
public class Solution {
    
    public int solve(int A, int B, int C) {
        return A/lcm(B,C);
    }
    
    public int gcd(int A, int B) {
        return B==0? A:gcd(B,A%B);
    }
    
    private int lcm(int A, int B){
        return (A/gcd(A,B))*B;
    }
}

File: 001-math/002-Divisor game/JS/Solution.js
function gcd(A , B){
	    return B==0?A:gcd(B,A%B);
}
	
function lcm(A , B){
         return ( Math.floor(A/gcd(A,B)))*B;
}
	
module.exports = { 
 //param A : integer
 //param B : integer
 //param C : integer
 //return an integer
	solve : function(A, B, C){
	    return  Math.floor(A/lcm(B,C));
	}
};
